lesson-4/README.md

```diff
@@ -0,0 +1,234 @@
+
+```bash
+kubectl create namespace monitoring
+```
+
+Now list all the namespaces via:
+```
+kubectl get namespaces
+NAME          STATUS    AGE
+default       Active    4m
+kube-public   Active    4m
+kube-system   Active    4m
+monitoring    Active    5s
+```
+As you can see, `monitoring` namespace was created successfully.
+
+### Deploying Prometheus
+
+NOTE: All `kubectl apply` commands should be run from the cloned workshop-k8s
+repository (see [Prerequisites]([Prerequisites](#Prerequisites)).
+
+#### ConfigMap
+Prometheus will get its configuration from a [Kubernetes ConfigMap](https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/).
+This allows us to update the configuration separate from the image.
+This is just a [Prometheus configuration](https://github.com/prometheus/prometheus/blob/master/documentation/examples/prometheus-kubernetes.yml)
+transformed to the Kubernetes manifest.
+
+To deploy this to Kubernetes run
+```bash
+kubectl apply -f prometheus-config.yaml
+```
+
+Make sure that the ConfigMap was created successfully:
+```bash
+kubectl get configmap prometheus-config -n monitoring
+NAME                DATA      AGE
+prometheus-config   1         25s
+```
+
+#### Prometheus
+
+Here is the Prometheus deployment that we're going to use during our workshop
+```yaml
+---
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRole
+metadata:
+  name: prometheus
+rules:
+- apiGroups: [""]
+  resources:
+  - nodes
+  - nodes/proxy
+  - services
+  - endpoints
+  - pods
+  verbs: ["get", "list", "watch"]
+- apiGroups:
+  - extensions
+  resources:
+  - ingresses
+  verbs: ["get", "list", "watch"]
+- nonResourceURLs: ["/metrics"]
+  verbs: ["get"]
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: default
+  namespace: monitoring
+---
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRoleBinding
+metadata:
+  name: prometheus
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: prometheus
+subjects:
+- kind: ServiceAccount
+  name: default
+  namespace: monitoring
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  labels:
+    name: prometheus
+  name: prometheus
+  namespace: monitoring
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      name: prometheus
+  strategy:
+    rollingUpdate:
+      maxSurge: 1
+      maxUnavailable: 1
+    type: RollingUpdate
+  template:
+    metadata:
+      creationTimestamp: null
+      labels:
+        name: prometheus
+      annotations:
+        prometheus.io/scrape: "true"
+        prometheus.io/port: "9090"
+    spec:
+      containers:
+      - args:
+        - --config.file=/etc/prometheus/prometheus.yml
+        - --storage.tsdb.path=/prometheus
+        - --storage.tsdb.retention=24h
+        command:
+        - /bin/prometheus
+        image: quay.io/prometheus/prometheus:v2.3.1
```

Comment:
@Evalle lets use our repo as well here, I've uploaded image to `docker.io/prgcont/prometheus:v2.3.1` Can we use that one?

Reply:
Good idea, let's use out image.